internal/job/job.go

```diff
@@ -0,0 +1,97 @@
+package job
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/wrboyce/jobsvc/internal/cgroup"
+)
+
+type Status string
+
+const (
+	StatusUnknown  Status = "unknown"
+	StatusRunning  Status = "running"
+	StatusFinished Status = "finished"
+	StatusFailed   Status = "failed"
+	StatusStopped  Status = "stopped"
+	StatusError    Status = "error"
+)
+
+type Job struct {
+	Id       string
+	Command  string
+	Args     []string
+	Status   Status
+	ExitCode int
+
+	Cmd *exec.Cmd
+}
+
+func NewJob(id, command string, args []string) (*Job, error) {
+	j := &Job{
+		Id:       id,
+		Command:  command,
+		Args:     args,
+		Status:   StatusUnknown,
+		ExitCode: -1,
+	}
+
+	if err := cgroup.Create(id); err != nil {
+		return nil, err
+	}
+
+	args = append([]string{id, command}, args...)
+	j.Cmd = exec.Command(os.Args[0], args...)
+	j.Cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	j.Cmd.Stdout = os.Stdout
+	j.Cmd.Stderr = os.Stderr
+
+	return j, nil
+}
+
+func (j *Job) Start() error {
+	if err := j.Cmd.Start(); err != nil {
+		j.Status = StatusError
+		return fmt.Errorf("failed to start job: %w", err)
+	} else {
+		j.Status = StatusRunning
+	}
+
+	go func() {
+		err := j.Cmd.Wait()
+		defer cgroup.Remove(j.Id)
+
+		if j.Status != StatusStopped {
+			if err != nil {
+				if exitErr, ok := err.(*exec.ExitError); ok {
```

Comment:
Prefer `errors.As` when checking for concrete errors.
```suggestion
				if exitErr := (*exec.ExitError)(nil); errors.As(err, &exitErr) {
```